### ISSUE TITLE
Refactor twisted.trial.test.test_assertions to separate synchronous from asynchronous

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5858]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5858|
|Type|enhancement|
|Created|2012-08-13 12:57:53Z|
|Branch|https://github.com/twisted/twisted/tree/sync-async-split-5858|

This test module covers a large portion of the helper methods provided by `TestCase`.  It tests both the synchronous helpers and the asynchronous helpers (such as `assertFailure` and `addCleanup`).

To facilitate splitting up the implementation of these two categories of features, it would help to split up the tests for them as well.

Split off of [#5853](https://github.com/twisted/twisted/issues/5853)

<details><summary>Searchable metadata</summary>

```
trac-id__5858 5858
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_sync_async_split_5858 branches/sync-async-split-5858
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1344862673000000 1344862673000000
changetime__1344885455000000 1344885455000000
version__None None
owner__exarkun exarkun
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [35167]) Branching to 'sync-async-split-5858'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [35168]) Factor sync and async tests into separate TestCases; also do a little more explicit bootstrapping.

refs [#5858](https://github.com/twisted/twisted/issues/5858)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun removed owner|
|-|-|

Made the described changes.  Note that I did a little bit of refactoring to split tests for particular assertion methods (eg `failIf`/`assertFalse`) into individual `TestCase` subclasses.  I didn't refactor all of the tests this way though, as that would have made the patch rather large.  I think that eventually this is what should be done to the tests, though.  If desired I can file a ticket for that when landing this branch (and add a note to the source referring to it).

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/sync-async-split-5858)

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)|@Julian set owner to @Julian|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_6"></a>|@Julian set owner to @exarkun|
|-|-|

Hi :)!

I think you have a build failure on 2.6, where the context manager form of `assertRaises` isn't present without using `unittest2`. Also, some of the docstrings on `TestSyncronousAssertions` are missing epytext markup in their docstrings, in case that matters.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [35173]) Avoid using unportable unittest.TestCase.assertRaises

refs [#5858](https://github.com/twisted/twisted/issues/5858)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun removed owner|
|-|-|

Thanks for the review, Julian!

I fixed the `assertRaises` issue in the revision linked above.  I'm not too concerned about the missing epytext, as it seems to largely be in methods I'm just moving around in the file, not otherwise modifying.  Updating them would be fine, but I'm not going to take that on as part of this ticket. :)

Thanks again for the review.

Latest build results [will be here](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/sync-async-split-5858).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_9"></a>|@itamarst set owner to @exarkun|
|-|-|

Looks good, merge if buildbot passes.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [35177]) Merge sync-async-split-5858 

Author: exarkun
Reviewer: Julian, itamar
Fixes: [#5858](https://github.com/twisted/twisted/issues/5858)

Divide the unit tests for most of the assertion helpers on `twisted.trial.unittest.TestCase`
into separate classes, one for synchronous helpers and another for asynchronous helpers, as
a step towards supporting a synchronous-only test case base class.
